up_esb/execution/executor.py

```diff
@@ -0,0 +1,322 @@
+"""Executor for executing tasks."""
+from threading import Condition, Lock, Thread
+from typing import NamedTuple
+
+import networkx as nx
+from unified_planning.plans import Plan, SequentialPlan, TimeTriggeredPlan
+from unified_planning.shortcuts import EndTiming, StartTiming, TimeInterval
+
+from up_esb.status import ActionNodeStatus, ConditionStatus
+
+# Action Result
+ActionResult = NamedTuple(
+    "ActionResult",
+    [
+        ("precondition_status", ConditionStatus),
+        ("action_status", ActionNodeStatus),
+        ("postcondition_status", ConditionStatus),
+        ("result", object),
+    ],
+)
+
+
+class TaskTracker:
+    """Track the amount of tasks that is being executed."""
+
+    def __init__(self):
+        self._lock = Lock()
+        self._condition = Condition(self._lock)
+        self._tasks = 0
+
+    def __enter__(self):
+        with self._lock:
+            self._tasks += 1
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        with self._lock:
+            self._tasks -= 1
+            self._condition.notify()
+
+    def wait(self):
+        """Wait until all tasks are finished."""
+        with self._lock:
+            while self._tasks > 0:
+                self._condition.wait()
+
+
+class TaskExecutor:
```

Comment:
It seems to me, that the `TaskExecutor` is not beeing used. Could be remove it or do you plan to use it in future developments?

Reply:
I plan to use it to manage time-triggered executions and parallel execution.